tools/tests/test_lint.py

```
@@ -130,73 +114,103 @@ def _ci(root: Path, text: str) -> None:
-    assert len(findings) == 1 and "emit_chartr.py" in findings[0]
+    assert len(findings) == 1 and expected in findings[0]
 
 
-def test_delivery_stays_quiet_on_a_wired_tree(tmp_path):
+def test_marketplace_source_reports_a_failed_read(tmp_path, monkeypatch):
     make_clean_tree(tmp_path)
-    assert lint.run(tmp_path) == []
+    manifest = tmp_path / ".claude-plugin" / "marketplace.json"
+    original_read_text = Path.read_text
+
+    def denied(path, *args, **kwargs):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `denied`

(ANN202)